src/firmware_new/boards/frankie_v1/BUILD

```diff
@@ -78,26 +77,118 @@ cc_library(
 )
 
 cc_library(
-    name = "frankie_v1_lwip",
+    name = "usb_otg",
     srcs = [
-        "ethernetif.c",
-        "lwip.c",
+        "usb_otg.c",
     ],
     hdrs = [
-        "FreeRTOSConfig.h",
-        "ethernetif.h",
-        "lwip.h",
-        "lwipopts.h",
+        "main.h",
+        "usb_otg.h",
     ],
     copts = [
+        # We don't want to mark warnings as errors for 3rd party code
+        "-Wno-error",
         "-DSTM32H743xx",
+    ],
+    restricted_to = ["//cc_toolchain:stm32h7"],
+    deps = [
+        ":hal",
+    ],
+    # This is _very_ important. If we don't do this, bazel will not link in the
+    # interrupt handling code, which *will not* cause an error, and *will* cause
+    # interrupts to stop working
+    alwayslink = True,
+)
+
+cc_library(
+    name = "crc",
+    srcs = [
+        "crc.c",
+    ],
+    hdrs = [
+        "crc.h",
+        "main.h",
+    ],
+    copts = [
         # We don't want to mark warnings as errors for 3rd party code
         "-Wno-error",
+        "-DSTM32H743xx",
```

Comment:
what does this do?

Reply:
just a macro that you need to set for HAL to work correctly